aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -526,18 +526,18 @@ int TensorIterator::num_reduce_dims() const {
     }                                                                     \
   }
 
-void TensorIterator::for_each(loop_t loop) {
-  for_each(LOOP_WRAPPER(ntensors(), loop));
+void TensorIterator::for_each(loop_t loop, int64_t grain_size) {
+  for_each(LOOP_WRAPPER(ntensors(), loop), grain_size);
 }
 
-void TensorIterator::for_each(loop2d_t loop) {
+void TensorIterator::for_each(loop2d_t loop, int64_t grain_size) {
   int64_t numel = this->numel();
   if (numel == 0) {
     return;
-  } else if (numel < internal::GRAIN_SIZE || at::get_num_threads() == 1) {
+  } else if (numel < internal::GRAIN_SIZE ||  at::get_num_threads() == 1) {
```

Comment:
Are you sure that you want to ignore passed grain_size here? If so, please add comment to the function explaining behaviour.

Reply:
If the number of element is less than the internal::GRAIN_SIZE or current run with pone single thread, no parallel version will be launched this no grain size needed here